include/seqan3/utility/views/zip.hpp

```diff
@@ -6,26 +6,476 @@
+template <bool Const>
+class zip_view<Views...>::sentinel
+{
+private:
+    constexpr explicit sentinel(zip::tuple_or_pair<
+                                    std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>...> end)
+    : end_(std::move(end))
+    {}
+
+    friend class zip_view<Views...>;
+public:
+    // Exposition-only. Is public such that it can be accessed by friends.
+    zip::tuple_or_pair<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>...> end_;
+
+    sentinel() = default;
+    constexpr sentinel(sentinel<!Const> i)
+        requires Const && (std::convertible_to<std::ranges::sentinel_t<Views>,
+                                               std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>> && ...)
+    : end_(std::move(i.end_))
+    {}
+
+    template <bool OtherConst>
+        requires (std::sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                                    std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr bool operator==(iterator<OtherConst> const & x, sentinel const & y)
+    {
+        return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            return ((std::get<N>(x.current_) == std::get<N>(y.end_)) || ...);
+        } (std::index_sequence_for<Views...>{});
+
+    }
+
+    template <bool OtherConst>
+        requires (std::sized_sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                                          std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr std::common_type_t<std::ranges::range_difference_t<view_helper::maybe_const<OtherConst, Views>>...>
+        operator-(iterator<OtherConst> const & x, sentinel const & y)
+    {
+        return [&] <size_t... N> (std::integer_sequence<size_t, N...>)
+        {
+            return std::ranges::min({(zip::abs(std::get<N>(x.current_) - std::get<N>(y.end_)), ...)});
+        } (std::index_sequence_for<Views...>{});
+    }
+
+    template <bool OtherConst>
+        requires (std::sized_sentinel_for<std::ranges::sentinel_t<view_helper::maybe_const<Const, Views>>,
+                  std::ranges::iterator_t<view_helper::maybe_const<OtherConst, Views>>> && ...)
+    friend constexpr std::common_type_t<std::ranges::range_difference_t<view_helper::maybe_const<OtherConst, Views>>...>
+        operator-(sentinel const & y, iterator<OtherConst> const & x)
+    {
+        return - (x - y);
+    }
+};
+
+// Here be dragons: Does not exactly follow the specifications on how to define a range adaptor (closure).
+// Seems to work.
+struct zip_fn
+{
+    template <typename ...rng_t>
+    constexpr auto operator()(rng_t&&... rngs) const
+    {
+        if constexpr (sizeof...(rng_t) == 0)
+            return std::views::empty<seqan3::common_tuple<>>;
+        else
+            return zip_view{std::forward<rng_t>(rngs)...};
+    }
```

Comment:
No "pipe syntax" for zip?

Reply:
As per gitter: Apparently it works, I'll add a unit test

Reply:
After testing: I added a proper adaptor :) 
Good catch!